psqlgraph/node.py

```diff
@@ -204,6 +215,59 @@ def edge_predicate(e):
                     queue.append((n, depth + 1))
                     marked.add(n.node_id)
 
+    def _dfs(self, edge_predicate=None, max_depth=None, edge_pointer="in"):
```

Comment:
Can you move this to a new module say `traversals`, this class is already bloated